analyzers/tests/SonarAnalyzer.UnitTest/TestCases/SymbolicExecution/Roslyn/NullPointerDereference.CSharp9.cs

```diff
@@ -116,15 +116,15 @@ public void PatternMatchingSwitch(object arg)
             string and not null => arg,
             _ => ""
         };
-        res.ToString();
+        res.ToString();     // Noncompliant FP FIXME
```

Comment:
I see some similar issues in my PR too:
https://github.com/SonarSource/sonar-dotnet/pull/6029/files#diff-bfec88ba4ce6ece6a162b6a907da17d9eedc5933614d8393152b4ea59fc5d73cL63-R69

This might be related to "not null" or other kinds of negations.

Reply:
I've added point to #5970 for me to investigate